lib/WorseReflection/Core/Inference/Resolver/ObjectCreationExpressionResolver.php

```diff
@@ -74,4 +78,17 @@ private function resolveClassType(NodeContextResolver $resolver, Frame $frame, O
         );
         return new GenericClassType($resolver->reflector(), $classType->name(), $templateMap->toArguments());
     }
+
+    private function resolveAnonymousClass(NodeContextResolver $resolver, Node $node): NodeContext
+    {
+        return NodeContextFactory::create(
+            'testing',
+            $node->getStartPosition(),
+            $node->getEndPosition(),
+            [
+                'symbol_type' => Symbol::CLASS_,
+                'type' => TypeFactory::reflectedClass($resolver->reflector(), NodeUtil::nameFromTokenOrNode($node, $node))
```

Comment:
again, i'm not sure how this works, and assume it works based on the anonymous class previously being parsed? we should avoid depending on the cache.

see https://github.com/Roave/BetterReflection/pull/300

probably the reflector should support accepting the `Node` instead of a "name".